script/mafia

```diff
@@ -0,0 +1,75 @@
+#!/bin/sh -eu
+
+fetch_latest () {
+  if [ -z ${MAFIA_TEST_MODE+x} ]; then
+    TZ=$(date +"%T")
+    curl --silent "https://raw.githubusercontent.com/ambiata/mafia/master/script/mafia?$TZ"
+  else
+    cat ../script/mafia
+  fi
+}
+
+run_upgrade () {
+  echo "Checking for a new version of mafia ..."
+  fetch_latest > /tmp/mafia
+
+  COMMIT_VERSION=$(git ls-remote https://github.com/ambiata/mafia | grep refs/heads/master | cut -f 1)
+  echo "# Version: $COMMIT_VERSION" >> /tmp/mafia
+
+  if ! cmp ./mafia /tmp/mafia >/dev/null 2>&1; then
+    echo "New version found and upgraded. You can now commit it to your git repo."
+    mv /tmp/mafia ./mafia
+    chmod +x ./mafia
+  else
+    echo "You have latest mafia script"
+  fi
+}
+
+exec_mafia () {
+  MAFIA_VERSION=$(awk '/^# Version: / { print $3; exit 0; }' $0)
+  MAFIA_BASE=$HOME/.ambiata/mafia/bin/mafia
+  MAFIA_PATH=$MAFIA_BASE-$MAFIA_VERSION
+
+  [ -f "$MAFIA_PATH" ] || {
+    MAFIA_TEMP=$(mktemp -d 2>/dev/null || mktemp -d -t 'exec_mafia')
+    trap "rm -rf \"$MAFIA_TEMP\"" EXIT
```

Comment:
Confirmed; I just tested that at least `/bin/sh` runs the command on SIGHUP, -INT and -TERM.